addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/BigRedirectsScanRule.java

```diff
@@ -70,22 +72,22 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
                 // Check if response is bigger than predicted
                 if (responseBodyLength > predictedResponseSize) {
                     // Response is larger than predicted so raise an alert
-                    newAlert()
-                            .setRisk(Alert.RISK_LOW)
-                            .setConfidence(Alert.CONFIDENCE_MEDIUM)
-                            .setDescription(getDescription())
-                            .setOtherInfo(
+                    createBigAlert(
                                     Constant.messages.getString(
                                             MESSAGE_PREFIX + "extrainfo",
                                             responseLocationHeaderURILength,
                                             locationHeaderValue,
                                             predictedResponseSize,
                                             responseBodyLength))
-                            .setSolution(getSolution())
-                            .setReference(getReference())
-                            .setCweId(201)
-                            .setWascId(13)
                             .raise();
+                } else {
+                    Pattern pattern = Pattern.compile("href", Pattern.CASE_INSENSITIVE);
+                    Matcher matcher = pattern.matcher(msg.getResponseBody().toString());
+                    long hrefCount = matcher.results().count();
+                    if (hrefCount > 1) {
+                        // Response more than one HREF so raise an alert
+                        createMultiAlert(String.valueOf(hrefCount)).raise();
+                    }
```

Comment:
Why not use the `source`? Otherwise the pattern should be compiled just once.

Reply:
This just seemed quicker than searching Source for all areas and a and ?? Then looking for hrefs.

Reply:
Fair, maybe use indexOf? That would be more efficient than using a pattern.

Reply:
On second thought it would be more work searching for different cases… pattern compiled once is good for me :)